### PR TITLE
Add DataLoader setup and config | Integrating EtageControl

### PR DESCRIPTION
**Pull request by Jose-Perigolo**
**DataLoader:**
- Setup and template
- Added a new main.js file to load any plugins/entity
- Added data-loader.ts file to receive any plugin from leaflet and load it.

**EtageControl:**
- Integration with pq-map.

**IN PROGRESS**

### CODE REVIEW

File: DataLoader/dev/main.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { DataLoader } from '@plantquest/data-loader';
+
+const map = L.map('map').setView([51.505, -0.09], 13);
+const dataLoader = new DataLoader({
+  query: {
+    project_id: 'project_id',
+    plant_id: 'plant_id',
+    stage: 'stage'
+  },
+  senecaPostFunction: SenecaPostFunction, // TODO: SENECA INTEGGRATION
```
Comment (rjrodger) [inline DataLoader/dev/main.js]: pass in the entire seneca object

File: DataLoader/src/data-loader.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  _isLoading: false,
+  _isLoaded: false,
+
+  initialize: function (options: DataLoaderOptions) {
+    L.Util.setOptions(this, options);
+    this._data = {};
+  },
+
+  loadData: async function () {
+    if (this._isLoading) {
```
Comment (rjrodger) [inline DataLoader/src/data-loader.ts]: remove this checking code with the timeouts, but keep the isLoading status flag - that should be public so remove the _

File: DataLoader/test/data-loader.test.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
+let Leaflet = require('../../verify/leaflet.js')
+let { PlantquestDataLoader } = require('../dist/etage-control.umd.cjs')
+
+describe('DataLoader', () => {
+  const options = {
```
Comment (rjrodger) [inline DataLoader/test/data-loader.test.js]: these are the options for Etage I think?

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: DataLoader/test/data-loader.test.js (hunk 3 above)
Comment (Jose-Perigolo) [reply DataLoader/test/data-loader.test.js]: Indeed. Not playing with tests yet. Corrected.

File: DataLoader/src/data-loader.ts (hunk 2 above)
Comment (Jose-Perigolo) [reply DataLoader/src/data-loader.ts]: Done!

File: DataLoader/dev/main.js (hunk 1 above)
Comment (Jose-Perigolo) [reply DataLoader/dev/main.js]: Done! Installed seneca lib locally for testing...

**Pull request merged**